### PR TITLE
attempt to fix https://github.com/arangodb/release-3.4/issues/96

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1615/

File: arangod/Replication/Syncer.cpp [hunk 1]
```diff
@@ -329,10 +329,20 @@ void Syncer::JobSynchronizer::request(std::function<void()> const& cb) {
     
 /// @brief notifies that a job was posted
 void Syncer::JobSynchronizer::jobPosted() {
-  CONDITION_LOCKER(guard, _condition);
-
-  TRI_ASSERT(_jobsInFlight == 0);
-  ++_jobsInFlight;
+  while (true) {
```
Comment (graetzer) [inline arangod/Replication/Syncer.cpp]: check is syncer is beeing stopped ?!
```suggestion
  while (!_syncer->isAborted()) {
```

File: arangod/Replication/Syncer.cpp [hunk 2]
```diff
@@ -341,6 +351,7 @@ void Syncer::JobSynchronizer::jobDone() {
 
   TRI_ASSERT(_jobsInFlight == 1);
   --_jobsInFlight;
+  _condition.signal();
```
Comment (graetzer) [inline arangod/Replication/Syncer.cpp]: we probably also need a way to signal the JobSynchronizer in the `setAborted(bool)` method ?

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1621/

**Pull request merged**